stdlib/3/tempfile.pyi

```diff
@@ -93,17 +92,48 @@ def NamedTemporaryFile(
 # It does not actually derive from IO[AnyStr], but it does implement the
 # protocol.
 class SpooledTemporaryFile(IO[AnyStr]):
-    def __init__(self, max_size: int = ..., mode: str = ...,
-                 buffering: int = ..., encoding: Optional[str] = ...,
-                 newline: Optional[str] = ..., suffix: Optional[str] = ...,
-                 prefix: Optional[str] = ..., dir: Optional[str] = ...
-                 ) -> None: ...
+    # bytes needs to go first, as default mode is to open as bytes
+    @overload
+    def __init__(
+        self: SpooledTemporaryFile[bytes],
+        max_size: int = ...,
+        mode: Literal["rb", "wb", "ab", "xb", "r+b", "w+b", "a+b", "x+b"] = ...,
+        buffering: int = ...,
+        encoding: Optional[str] = ...,
+        newline: Optional[str] = ...,
+        suffix: Optional[str] = ...,
+        prefix: Optional[str] = ...,
+        dir: Optional[str] = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self: SpooledTemporaryFile[str],
+        max_size: int = ...,
+        mode: Literal["r", "w", "a", "x", "r+", "w+", "a+", "x+", "rt", "wt", "at", "xt", "r+t", "w+t", "a+t", "x+t"] = ...,
```

Comment:
Also just `""`.

Reply:
Same error as above on Python 3.7.4, Python 2.7.16 has it's own `ValueError: empty mode string` error. It is only seen after the SpooledTemporaryFile creates a `TemporaryFile` internally, which can be forced using `~SpooledTemporaryFile.rollover()`.

```python
import tempfile

with tempfile.SpooledTemporaryFile(mode="") as tmpfile:
    tmpfile.rollover() # empty mode fails on this line, not at constructor
```